Module not found when using index.tsx in root

I've run into another issue using the TypeScript example with the latest react-static. If I change `index.js` to `index.tsx` I get the following error:

```bash
Failed to rebuild.
multi ./node_modules/react-hot-loader/patch.js ./node_modules/react-dev-utils/webpackHotDevClient.js (webpack)/hot/only-dev-server.js ./src/index.js
Module not found: Can't resolve 'path\to\src\index.js' in 'path\to\node_modules\react-static\lib\node_modules'
```

I feel like this is another example of an issue that may have been flushed out earlier if the TS example wasn't a hybrid of `js` and `tsx` files. I understand it's possible to mix them, but I'm not sure it's the most common scenario, especially for a new TypeScript project.

--- a/examples/typescript/index.d.ts
+++ b/examples/typescript/index.d.ts
@@ -1,0 +1,1 @@
+declare module '*.png'

--- a/examples/typescript/package.json
+++ b/examples/typescript/package.json
@@ -22,10 +22,16 @@
     "@types/react-dom": "^16.0.2",
     "@types/react-hot-loader": "^3.0.5",
     "@types/react-router": "^4.0.16",
+    "@types/webpack-env": "^1.13.3",
     "convert-tsconfig-paths-to-webpack-aliases": "^0.9.2",
     "eslint-config-react-tools": "1.x.x",
     "serve": "^6.4.1",
     "ts-loader": "^3.1.1",
     "typescript": "^2.6.1"
+  },
+  "prettier": {
+    "semi": false,
+    "singleQuote": true,
+    "trailingComma": "all"
   }
 }

--- a/examples/typescript/src/App.tsx
+++ b/examples/typescript/src/App.tsx
@@ -1,6 +1,5 @@
 import React from 'react'
 import { Router, Link } from 'react-static'
-//
 import Routes from 'react-static-routes'
 
 import './app.css'

--- a/examples/typescript/src/app.css
+++ b/examples/typescript/src/app.css
@@ -1,6 +1,6 @@
 body {
-  font-family: 'HelveticaNeue-Light', 'Helvetica Neue Light', 'Helvetica Neue', Helvetica, Arial,
-    'Lucida Grande', sans-serif;
+  font-family: 'HelveticaNeue-Light', 'Helvetica Neue Light', 'Helvetica Neue',
+    Helvetica, Arial, 'Lucida Grande', sans-serif;
   font-weight: 300;
   font-size: 16px;
   margin: 0;

--- a/examples/typescript/src/containers/404.tsx
+++ b/examples/typescript/src/containers/404.tsx
@@ -1,5 +1,4 @@
 import React from 'react'
-//
 
 export default () => (
   <div>

--- a/examples/typescript/src/containers/About.tsx
+++ b/examples/typescript/src/containers/About.tsx
@@ -3,12 +3,14 @@ import React from 'react'
 // Example of using paths alias via tsconfig.json to easily re-use legacy code
 // Also, use TypeScript and JavaScript interchangeably.
 import aliasExample from '@myPathAlias/component'
-//
 
 export default () => (
   <div>
     <h1>This is what we're all about.</h1>
-    <p>React, static sites, performance, speed. It's the stuff that makes us tick.</p>
+    <p>
+      React, static sites, performance, speed. It's the stuff that makes us
+      tick.
+    </p>
     {aliasExample}
   </div>
 )

--- a/examples/typescript/src/containers/Blog.tsx
+++ b/examples/typescript/src/containers/Blog.tsx
@@ -1,9 +1,12 @@
-
 import React from 'react'
 import { getRouteProps, Link } from 'react-static'
-//
+import { Post } from '../types'
 
-export default getRouteProps(({ posts }) => (
+interface Props {
+  posts: Post[]
+}
+
+export default getRouteProps(({ posts }: Props) => (
   <div>
     <h1>It's blog time.</h1>
     <br />

--- a/examples/typescript/src/containers/Home.tsx
+++ b/examples/typescript/src/containers/Home.tsx
@@ -1,6 +1,6 @@
 import React from 'react'
 import { getSiteProps } from 'react-static'
-//
+
 import logoImg from '../logo.png'
 
 export default getSiteProps(() => (

--- a/examples/typescript/src/containers/Post.tsx
+++ b/examples/typescript/src/containers/Post.tsx
@@ -1,6 +1,6 @@
 import React from 'react'
 import { getRouteProps, Link } from 'react-static'
-import { Post } from 'src/types'
+import { Post } from '../types'
 
 interface Props {
   post: Post

--- a/examples/typescript/src/containers/Post.tsx
+++ b/examples/typescript/src/containers/Post.tsx
@@ -1,8 +1,12 @@
 import React from 'react'
 import { getRouteProps, Link } from 'react-static'
-//
+import { Post } from 'src/types'
 
-export default getRouteProps(({ post }) => (
+interface Props {
+  post: Post
+}
+
+export default getRouteProps(({ post }: Props) => (
   <div>
     <Link to="/blog/">{'<'} Back</Link>
     <br />

--- a/examples/typescript/src/index.tsx
+++ b/examples/typescript/src/index.tsx
@@ -11,7 +11,7 @@ export default App
 // Render your app
 if (typeof document !== 'undefined') {
   const renderMethod = module.hot ? ReactDOM.render : ReactDOM.hydrate
-  const render = Comp => {
+  const render = (Comp: React.ComponentType) => {
     renderMethod(
       <AppContainer>
         <Comp />

--- a/examples/typescript/src/path-alias-example/component.tsx
+++ b/examples/typescript/src/path-alias-example/component.tsx
@@ -1,5 +1,4 @@
 import React from 'react'
-//
 
 // This is referenced in src/containers/About.tsx as an alias example
 // You can change the alias to this path inside tsconfig.json -> paths
@@ -7,8 +6,8 @@ import React from 'react'
 
 const helloFromThePast = (
   <h4>
-    TypeScript is enabled. Use it when and where you want. This text was imported via a TypeScript
-    alias.
+    TypeScript is enabled. Use it when and where you want. This text was
+    imported via a TypeScript alias.
   </h4>
 )
 

--- a/examples/typescript/src/types.ts
+++ b/examples/typescript/src/types.ts
@@ -1,0 +1,5 @@
+export interface Post {
+  body: string
+  id: number
+  title: string
+}

--- a/examples/typescript/static.config.js
+++ b/examples/typescript/static.config.js
@@ -1,14 +1,18 @@
 import axios from 'axios'
+import path from 'path'
 
 // Paths Aliases defined through tsconfig.json
 const typescriptWebpackPaths = require('./webpack.config.js')
 
 export default {
+  entry: path.join(__dirname, 'src', 'index.tsx'),
   getSiteProps: () => ({
     title: 'React Static',
   }),
   getRoutes: async () => {
-    const { data: posts } = await axios.get('https://jsonplaceholder.typicode.com/posts')
+    const { data: posts } = await axios.get(
+      'https://jsonplaceholder.typicode.com/posts',
+    )
     return [
       {
         path: '/',

--- a/examples/typescript/tsconfig.json
+++ b/examples/typescript/tsconfig.json
@@ -29,5 +29,8 @@
       "es2016"
     ]
   },
-  "exclude": ["dist/**/*", "public/**/*"]
+  "exclude": [
+    "dist/**/*",
+    "public/**/*"
+  ]
 }

